src/Illuminate/Console/GeneratorCommand.php

```diff
@@ -205,6 +205,59 @@ protected function replaceClass($stub, $name)
         return str_replace('DummyClass', $class, $stub);
     }
 
+    /**
+     * Replace the parent class for the given stub.
+     *
+     * @param  string  $stub
+     * @param  string  $name
+     * @return $this
+     */
+    protected function replaceParent(&$stub, $name)
+    {
+        $parent = $this->getParent() ?? $this->defaultParent ?? '';
+
+        $replace = [];
+
+        if ($this->getNamespace($parent) === $this->getNamespace($name) || $parent === '') {
+            $replace['use NamespacedDummyParent;'.PHP_EOL] = '';
```

Comment:
inspired by ControllerMakeCommand Line 96 `$replace["use {$controllerNamespace}\Controller;\n"] = '';`
I noticed that \n doesn't work on Windows and there are no tests written for this behavior to detect it, on Windows replacing it with \r\n works, but that's why PHP_EOL exists and it works (did not test on other operating systems).